### ISSUE TITLE
Vib module repository 

### ISSUE THREAD

**Issue by arejula27** [Vanilla-OS/Vib#11]
I see Vib as an amazing tool, not just for building "official" images but also for creating custom images for individuals with limited knowledge about the internals of the OS. I appreciate its functionality, as seen, for instance, in the [core-image](https://github.com/Vanilla-OS/core-image/tree/main/modules) repository. However, using it requires knowledge of the specific packages to use and the installation process.

I propose the creation of a repository to host metadata about packages, and possibly modules (even though just the URL could suffice), similar to platforms like [Flathub](https://flathub.org/about) or [Eratosthenes](https://packages.vanillaos.org/). This way, individuals seeking specific components, such as Nvidia drivers or [pipewire](https://github.com/Vanilla-OS/core-image/blob/main/modules/60-sound.yml), can visit the website, consult the module reference, and integrate it into their recipes:
```
 name: packages-modules
  type: includes
  includes:
    - https://repo.com/module/sound.yaml
```

This approach will not only simplify the image-building process but also foster community growth. For instance, someone who successfully configures wifi drivers for an unusual laptop (e.g., Mac) can share their solution, allowing others to utilize the module and contributing to collaborative problem-solving.

**Comment by xynydev** [comment]
This feature could be similar to how custom actions work in Github, where one could include a module from a hosted git repository, ideally with one repository being able to include multiple modules.

**Comment by mirkobrombin** [comment]
I like the idea

**Comment by axtloss** [comment]
Vib supports fetching modules from git repositories [as seen in the example](https://github.com/Vanilla-OS/Vib/blob/main/example/example.yml#L49) (though i don't see it in the documentation, would have to add that), so this could be easily offered by simply having a git repo with vib modules to use

**Comment by mirkobrombin** [comment]
> Vib supports fetching modules from git repositories [as seen in the example](https://github.com/Vanilla-OS/Vib/blob/main/example/example.yml#L49) (though i don't see it in the documentation, would have to add that), so this could be easily offered by simply having a git repo with vib modules to use

It's documented here https://docs.vanillaos.org/vib/en/project-structure probably deserves a dedicated article

**Comment by arejula27** [comment]
If implementation is required pls notify me, I would like to help

**Comment by axtloss** [comment]
There isn't really anything regarding code to be done in vib, as it supports remote modules already. 
What we need is a main repo (if we as vanilla even want to make a central vib module repo) and maybe a tool similiar to Atlas that lets people browse vib module repos

**Comment by arejula27** [comment]
> There isn't really anything regarding code to be done in vib, as it supports remote modules already. What we need is a main repo (if we as vanilla even want to make a central vib module repo) and maybe a tool similiar to Atlas that lets people browse vib module repos

How about enabling users to share their module/recipe files? This would enable vendors to export their configurations and drivers so that people can install them alongside the official module. For instance, if I want to use a new printer, I can simply go to the repository, search for the company's printer module, and add it to my recipe.

This would also be beneficial for sharing custom configurations. For example, a few years ago, I had difficulty getting my Mac's camera and Wi-Fi to work. Instead of copying a bunch of commands from Reddit, I could easily add the solution to my recipe.

**Comment by taukakao** [comment]
One problem to always keep in mind is security. 
It should be clear to any user that checking modules before adding them to an image is crucial. 
Modules can always contain malware.

**Comment by axtloss** [comment]
> How about enabling users to share their module/recipe files?

that wouldn't be something for vib to handle.

To combat the issue that tau described it shouldn't be possible to fetch modules without reading them. the best way to ensure this is to make it not possible for vib to automatically pull modules from a central repository but require the user to manually add the link in their module, to get the link they would most likely have the entire module directly visible at the module at least once (assuming GitHub or other git forges are used).

Basically we won't implement something similar to the aur, as it's a complete security nightmare.

**Comment by axtloss** [comment]
We now have a shared module repository here:
https://github.com/Vanilla-OS/vib-shared-modules

each new module will go through manual verification by me, to ensure that modules are not malicious, and most modules will have to be maintained by the person who added them, unless they are popular, in which case I may assist with maintenance.
